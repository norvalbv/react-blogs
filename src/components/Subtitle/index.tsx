import useGetClassName from 'hooks/useGetClassName';
import { ReactElement } from 'react';
import { isLabelInProps } from 'utils';

export type Props = {
  children?: string | ReactElement;
  className?: string;
  props?: unknown;
};

const Subtitle = ({ children, className, ...props }: Props): ReactElement | null => {
  const { getClassName } = useGetClassName();

  if (!children) {
    return null;
  }

  if (typeof children === 'string') {
    return (
      <h4 className={getClassName({ tag: 'h4', className })} {...isLabelInProps(props)}>
        {children}
      </h4>
    );
  }

  return children;
};

export default Subtitle;
