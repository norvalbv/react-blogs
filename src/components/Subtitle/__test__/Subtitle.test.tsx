import { FC } from 'react';
import renderer from 'react-test-renderer';

import Subtitle, { Props } from '..';

const TitleWithAllProps: FC<Required<Props>> = Subtitle;

describe('<Subtitle />', () => {
  test('should render nothing if subtitle is not provided', () => {
    const tree = renderer.create(<Subtitle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render correctly with all props', () => {
    const tree = renderer
      .create(
        <TitleWithAllProps
          props={{ testId: 'Test Id' }}
          theme={{
            theme: undefined,
            code: undefined,
            overrides: undefined,
          }}
          className="text-sm"
        >
          Test subtitle
        </TitleWithAllProps>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
