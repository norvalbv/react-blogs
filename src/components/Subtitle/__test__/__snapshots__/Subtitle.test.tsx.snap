// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Subtitle /> should render correctly with all props 1`] = `
<h4
  className="text-sm"
  props={
    {
      "testId": "Test Id",
    }
  }
>
  Test subtitle
</h4>
`;

exports[`<Subtitle /> should render nothing if subtitle is not provided 1`] = `null`;
